core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ReplicateJoinAndSemiJoinInDelete.java

```diff
@@ -69,7 +71,7 @@ public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Void> context)
             SemiJoinNode rewrittenNode = (SemiJoinNode) node.replaceChildren(ImmutableList.of(sourceRewritten, filteringSourceRewritten));
 
             if (isDeleteQuery) {
-                return rewrittenNode.withDistributionType(SemiJoinNode.DistributionType.REPLICATED);
```

Comment:
revert. it's better to qualify both enums now that we use `SemiJoinNode.DistributionType.REPLICATED` and `JoinNode.DistributionType.REPLICATED` within this class.